src/pipecat/transports/services/pipecat_cloud.py

```diff
@@ -0,0 +1,313 @@
+        room_url: Optional[str] = None,
+        token: Optional[str] = None,
+        webrtc_connection: Optional[SmallWebRTCConnection] = None,
+        session_id: Optional[str] = None,
+        body: Optional[dict] = None,
+    ):
+        """Initialize session arguments for any supported transport type."""
+        if websocket is not None:
+            self._args = WebSocketSessionArguments(websocket=websocket, session_id=session_id)
+        elif room_url is not None:
+            self._args = DailySessionArguments(
+                # TODO-CB: bot_name is missing from DailySessionArguments?
+                room_url=room_url,
+                token=token,
+                session_id=session_id,
+                body=body,
+            )
+        elif webrtc_connection is not None:
+            self._args = WebRTCSessionArguments(
+                webrtc_connection=webrtc_connection, session_id=session_id
+            )
+        else:
+            raise ValueError(
+                "Must provide either websocket, (room_url, token, bot_name), or webrtc_connection"
+            )
+
+    @property
+    def args(self):
+        """Get the underlying session arguments."""
+        return self._args
+
+
+class PipecatCloudTransport(BaseTransport):
+    """A transport that wraps FastAPIWebsocketTransport, SmallWebRTCTransport, and DailyTransport.
+
+    This transport will instantiate one of the three underlying transports based on the
+    session arguments provided to the constructor.
+
+    Event handlers:
+        @event_handler("on_client_connected"): Called when a client connects. Maps to:
+            - FastAPIWebsocketTransport.event_handler("on_client_connected")
+            - SmallWebRTCTransport.event_handler("on_client_connected")
+            - DailyTransport.event_handler("on_first_participant_joined")
+
+        @event_handler("on_client_disconnected"): Called when a client disconnects. Maps to:
+            - FastAPIWebsocketTransport.event_handler("on_client_disconnected")
+            - SmallWebRTCTransport.event_handler("on_client_disconnected")
+            - DailyTransport.event_handler("on_participant_left")
+
+        Other event handlers are passed through directly to the underlying transport.
+
+    Args:
+        session_args: Arguments for creating the session. The type of arguments determines
+            which transport will be used.
+        params: Configuration parameters for the transport. Parameters will be extracted
+            based on the session arguments type.
+        input_name: Optional name for the input transport.
+        output_name: Optional name for the output transport.
+    """
+
+    # Event name mappings for each transport type
+    # Only include events that need to be mapped differently
+    _EVENT_MAPPINGS = {
+        DailyTransport: {
+            "on_client_connected": "on_first_participant_joined",
+            "on_client_disconnected": "on_participant_left",
+        },
+    }
```

Comment:
I'm not sure this would work well. Each event has different number of arguments. For example, `on_participant_left` has 2, the participant and the reason. And websocket `on_client_disconnected` only has one. So you would end up with python errors.

Reply:
One option would be to do this per event. So, this class would have the common events we want to expose and it would register an event handler with the internal transport. Mainly it would acats as a proxy. Something like:

```
self._register_handler("on_client_connected")

@self._transport.event_handler("on_participant_left"):
async def on_participant_left(self, participant, reason):
    self._call_event_handler("on_client_disconnected", participant)
```

That said, the user then will not no what is the first argument. It might change between transports...